p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -2647,196 +2645,191 @@ X.v1  // reference to v1
 v1    // error - v1 is not in the top-level namespace
 ~ End P4Example
 
-An ```enum``` value can only be compared for equality/difference using ```== and !=```. ```enum```
-values cannot be cast to or from any other types.
+Similar to errors, ```enum``` expressions only support equality (`==`)
+and inequality (`!=`) comparisons. Expressions whose type is an `enum`
+cannot be cast to or from any other type.
 
-When ```enum``` values appear in the control-plane API the compiler
-back-end has to choose a suitable serialization data type and
+Note that if an ```enum``` value appears in the control-plane API, the
+compiler must select a suitable serialization data type and
```

Comment:
It is fundamental, enums are completely abstract.
How would you propose casting enums and to what?